adapters/Mintegral/MintegralAdapter/Bidding/GADMAdapterMintegralNativeAdLoader.m

```diff
@@ -0,0 +1,284 @@
+            _adLoadCompletionHandler(nil,error);
+        }
+    }
+}
+
+- (void)nativeAdsFailedToLoadWithError:(nonnull NSError *)error bidNativeManager:(nonnull MTGBidNativeAdManager *)bidNativeManager {
+    if (_adLoadCompletionHandler) {
+        _adLoadCompletionHandler(nil,error);
+    }
+}
+
+- (void)nativeAdDidClick:(nonnull MTGCampaign *)nativeAd bidNativeManager:(nonnull MTGBidNativeAdManager *)bidNativeManager {
+    [_adEventDelegate reportClick];
+}
+
+- (void)nativeAdImpressionWithType:(MTGAdSourceType)type bidNativeManager:(nonnull MTGBidNativeAdManager *)bidNativeManager {
+    [_adEventDelegate reportImpression];
+}
+
+#pragma mark - MTGMediaViewDelegate
+- (void)MTGMediaViewWillEnterFullscreen:(MTGMediaView *)mediaView {
+    [_adEventDelegate willPresentFullScreenView];
+}
+
+- (void)MTGMediaViewDidExitFullscreen:(MTGMediaView *)mediaView {
+    [_adEventDelegate didDismissFullScreenView];
+}
+
+- (void)MTGMediaViewVideoDidStart:(MTGMediaView *)mediaView {
+    [_adEventDelegate didPlayVideo];
+}
+
+- (void)MTGMediaViewVideoPlayCompleted:(MTGMediaView *)mediaView {
+    [_adEventDelegate didEndVideo];
+}
+
+- (void)nativeAdDidClick:(nonnull MTGCampaign *)nativeAd mediaView:(MTGMediaView *)mediaView {
+    [_adEventDelegate reportClick];
+}
+
+- (void)nativeAdImpressionWithType:(MTGAdSourceType)type mediaView:(MTGMediaView *)mediaView {
+    [_adEventDelegate reportImpression];
+}
+
+#pragma mark - GADMediatedUnifiedNativeAd
+- (NSString *)headline {
+    return _campaign.appName;
+}
+
+- (NSArray *)images {
+    return nil;
+}
+
+- (NSString *)body {
+    return _campaign.appDesc;
+}
+
+- (GADNativeAdImage *)icon {
+    return _icon;
+}
+
+- (NSString *)callToAction {
+    return _campaign.adCall;
+}
+
+- (NSDecimalNumber *)starRating {
+    NSString *star = [NSString stringWithFormat:@"%@",[_campaign valueForKey:@"star"]];
+    return [NSDecimalNumber decimalNumberWithString:star];
+}
+
+- (NSString *)store {
+    return nil;
+}
+
+- (NSString *)price {
+    return nil;
+}
+
+-(NSString *)advertiser{
+    return nil;
+}
+
+- (NSDictionary *)extraAssets {
+    return nil;
+}
+
+- (BOOL)hasVideoContent {
+    return YES;
+}
+
+-  (UIView *)mediaView{
+    [_mediaView setMediaSourceWithCampaign:_campaign unitId:_adUnitId];
+    return _mediaView;
+}
+
+- (UIView *)adChoicesView {
+    return _adChoicesView;
+}
+
+#pragma mark - GADMediationNativeAd
+- (BOOL)handlesUserClicks{
+    return YES;
+}
+
+- (BOOL)handlesUserImpressions{
+    return YES;
+}
+
+#pragma mark - GADMediatedUnifiedNativeAd
+-(void)didRenderInView:(UIView *)view clickableAssetViews:(NSDictionary<GADNativeAssetIdentifier,UIView *> *)clickableAssetViews nonclickableAssetViews:(NSDictionary<GADNativeAssetIdentifier,UIView *> *)nonclickableAssetViews viewController:(UIViewController *)viewController{
+    [_nativeManager registerViewForInteraction:view withClickableViews:clickableAssetViews.allValues withCampaign:_campaign];
```

Comment:
I saw the feedback that the Android native ad icon and background can be clicked is incorrect. Does iOS need to pay attention to this problem? The value I pass in this method is clickableAssetViews, but I found that the icon and title are clickable, and I found that the clickableAssetViews parameter contains all the view elements. I see that pangle native ad is also implemented like this.
[https://github.com/googleads/googleads-mobile-ios-mediation/blob/4d354d192e2f831541b35894952f266b2f91b1e9/adapters/Pangle/PangleAdapter/Bidding/GADPangleRTBNativeRenderer.m#L212](url)

Does MintegralAdSDK need to be controlled if the icon is not clickable? Or do I just need to focus on the clickableAssetViews parameter?

If I need MintegralAdSDK control, how should I do it?

Thanks!

Reply:
@Mintegral-Lee 
To clarify, the icon image (e.g. app store image or advertiser logo) and title can be clickable, but the background and the ad attribution (i.e. "Ad") can't be clickable. The AdChoices icon can be clickable but it should redirect users to Mintegral's privacy page and the click shouldn't be counted.


You can read more in [AdMob native ads policy compliance checklist](https://support.google.com/admob/answer/6240814?hl=en&ref_topic=7384666).

Reply:
OK, thanks!
MintegralAdSDK can only click icon image, CTA, title, ad choices will only open the Mintegral privacy page.

Reply:
@Mintegral-Lee 
Sorry, that native ad policy compliance checklist is actually not strictly enforced on third party (i.e. Mintegral) ads. Therefore I will defer to Mintegral team on whether which areas are clickable.

Please let me know if you would like to update the adapters.

Reply:
Confirmed by Karen that we can keep it as is.